webapp/packages/plugin-data-import/src/DataImportDialog/DataImportDialog.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observer } from 'mobx-react-lite';
+
+import {
+  Button,
+  CommonDialogBody,
+  CommonDialogFooter,
+  CommonDialogHeader,
+  CommonDialogWrapper,
+  Container,
+  Fill,
+  useTranslate,
+} from '@cloudbeaver/core-blocks';
+import type { DialogComponent } from '@cloudbeaver/core-dialogs';
+
+import { DataImportFileUploader } from './DataImportFileUploader';
+import { EDataImportDialogStep } from './EDataImportDialogStep';
+import type { IDataImportDialogState } from './IDataImportDialogState';
+import { ImportProcessorList } from './ImportProcessorList';
+import { useDataImportDialog } from './useDataImportDialog';
+
+export interface IDataImportDialogResult {
+  file: File;
+  processorId: string;
+}
+
+export interface IDataImportDialogPayload {
+  tableName: string;
+  initialState?: IDataImportDialogState;
+}
+
+export const DataImportDialog: DialogComponent<IDataImportDialogPayload, IDataImportDialogResult> = observer(function DataImportDialog({
+  payload,
+  resolveDialog,
+  rejectDialog,
+}) {
+  const translate = useTranslate();
+  const dialog = useDataImportDialog(payload.initialState);
+
+  function handleImport() {
+    if (!dialog.state.file) {
+      throw new Error('File must be provided');
+    }
+
+    if (!dialog.state.selectedProcessor) {
+      throw new Error('Processor must be provided');
+    }
```

Comment:
why not use Form + useCustomInputValidation (or maybe you can use HTML5 validation props on inputs, like required)

Reply:
Its not validation, its types, you cant submit form without file